### DISCUSSION TITLE
task_id must not be empty on chain of groups

### DISCUSSION THREAD

**Discussion by cadlagtrader** [celery/celery#8578] [Q&A] [answered]
Hello

I am having an issue when wrapping groups into chains, when one of the task in a group fail and more than one task remain in the chain

`ValueError: task_id must not be empty. Got None instead.`

Below is the minimal code to trigger the behavior. Any hint ? Or way to circumvent the issue ? I have tried to combine on_error / link_error, but this ValueError always go through

```
import time
from celery import Celery, chain, group



app = Celery("celery_failing", broker="redis://localhost:6379", backend="redis://localhost:6379")


@app.task
def add(x, y):
    time.sleep(3)
    return x + y


@app.task
def failc():
    raise Exception("failc")


if __name__ == "__main__":

    mychain = chain(group([add.si(15,7), failc.si()]), add.si(9,7), add.si(5,7))
    ret = mychain.apply_async()
    

    while not ret.ready():
        time.sleep(1)
        print(".")
```

**Comment by anzhurov** [comment]
@cadlagtrader any success on this ? have the same problem :c

**Comment by Nusnus** [comment, marked as answer]
@cadlagtrader @anzhurov 
Fixed in `main`:
* #8798

**Reply by Nusnus** [reply]
Will be available in the upcoming releases for v5.4.0 of Celery, or already if I'm not mistaken in v5.4.0rc1.

**Reply by anzhurov** [reply]
Thank you very much! When do you plan the release ?

**Reply by Nusnus** [reply]
> Thank you very much! When do you plan the release ?
> 
> 

Late March/start of April.
RC2 will be released probably a week or two before.

Glad I could help ❤️

**Comment by cadlagtrader** [comment]
Thank you @Nusnus !

**Comment by auvipy** [comment]
Got another fix here https://github.com/celery/celery/pull/9774
